### PR TITLE
Basic diff visualisation for plain text edits

### PR DESCRIPTION
**Pull request by bwindels**
Part of https://github.com/vector-im/riot-web/issues/9493

image

### CODE REVIEW

File: src/components/views/messages/EditHistoryMessage.js [hunk 1]
```diff
@@ -115,16 +126,36 @@ export default class EditHistoryMessage extends React.PureComponent {
         );
     }
 
+    _diffIt(oldBody, newBody) {
```
Comment (jryans) [inline src/components/views/messages/EditHistoryMessage.js]: Nit: Maybe just `_diff`? Unsure why this method would use an `It` suffix...

File: src/components/views/messages/EditHistoryMessage.js [hunk 2]
```diff
@@ -25,6 +25,17 @@ import sdk from '../../../index';
...
+    return originalContent["m.new_content"] || originalContent;
+}
+
+function isPlainMessage(event) {
+    const content = getReplacedContent(event);
+    return content.msgtype === "m.text" && !content.format;
+}
 
 export default class EditHistoryMessage extends React.PureComponent {
     static propTypes = {
```
Comment (jryans) [inline src/components/views/messages/EditHistoryMessage.js]: Please add missing prop types.

File: package.json [hunk 3]
```diff
@@ -66,6 +66,7 @@
     "classnames": "^2.1.2",
     "commonmark": "^0.28.1",
     "counterpart": "^0.18.0",
+    "diff-match-patch": "^1.0.4",
```
Comment (jryans) [inline package.json]: Is this a big dependency by any chance?

File: src/components/views/messages/EditHistoryMessage.js (hunk 1 above)
Comment (bwindels) [reply src/components/views/messages/EditHistoryMessage.js]: Whoops, forgot to rename after prototype code :)

File: package.json (hunk 3 above)
Comment (bwindels) [reply package.json]: It's 75kb of JS.

Comment (jryans) [reply package.json]: Okay, I assume that's not _too_ much in the scheme of things...

**Pull request merged**